windows-agent/internal/daemon/daemon.go

```diff
@@ -225,9 +237,29 @@ func (d *Daemon) serve(ctx context.Context, opts *options) (err error) {
 	wslNetAvailable := true
 	wslIP, err := getWslIP(ctx, *opts)
 	if err != nil {
-		log.Warningf(ctx, "could not get the WSL adapter IP: %v", err)
 		wslNetAvailable = false
 		wslIP = net.IPv4(127, 0, 0, 1)
+
+		log.Warningf(ctx, "Daemon: could not get the WSL adapter IP: %v. Starting network monitoring", err)
+		n, err := subscribe(ctx, func(added []string) bool {
+			for _, adapter := range added {
+				if strings.Contains(adapter, "(WSL") {
+					log.Warningf(ctx, "Daemon: new adapter detected: %s", adapter)
+					d.restart(ctx)
+					return false
+				}
+			}
+
+			// Not found yet, let's keep monitoring.
+			return true
+		}, opts)
+
+		if err != nil {
+			log.Errorf(ctx, "Daemon: could not start network monitoring: %v", err)
+			// should we return (and not proceed with serving) instead?
```

Comment:
I think we should if we can’t listen to the adapter and are only listening on the local port as it’s not going to be fixed magically and we just defer the error to later (when we install the first distro)